theano/tensor/opt.py

```diff
@@ -3052,9 +3122,52 @@ def local_remove_switch_const_cond(node):
                 out = T.alloc(out, *[node.outputs[0].shape[i] for i
                                      in xrange(out.ndim)])
             return [out]
-
+        # if left is right -> left
+        if node.inputs[1] is node.inputs[2]:
+            return [node.inputs[1]]
+        # if left equal right -> left
+        if (T.extract_constant(node.inputs[1]) ==
+            T.extract_constant(node.inputs[2])):
```

Comment:
I have hypothesis why  this case isn't covered by 3126, but where did you decide to add this case? It can be fixed by other ways, like doing the merge optimizer more frequently.